To-do list/Controllers/TaskDetailsViewController.swift

```diff
@@ -7,91 +7,42 @@
 
 import UIKit
 
+protocol TaskDetailsViewControllerDelegate: AnyObject {
+    func taskDetails(_ controller: TaskDetailsViewController, didCreateUpdate task: Task)
+}
+
 class TaskDetailsViewController: UIViewController {
     
+    weak var delegate: TaskDetailsViewControllerDelegate?
+    
     @IBOutlet weak var titleTextField: UITextField!
     @IBOutlet weak var descriptionTextField: UITextField!
     @IBOutlet weak var statusButtonOutlet: UIButton!
+  
+    enum ViewType {
+        case createTask
+        case updateTask
+    }
     
-    var users = [User]()
-    var userIndex = Int()
-    var taskIndex = Int()
-    var counter = 0
-    var status = "to do"
-    var titleField = ""
-    var choosenTableViewCell = Bool()
+    var task = Task()
+    var viewType = ViewType.createTask
+    var id = Int()
```

Comment:
you don't need id variable, because you already have `task` variable with `id` property

Reply:
done